packages/volto-slate/src/editor/plugins/StyleMenu/utils.js

```diff
@@ -72,20 +72,29 @@ export const toggleInlineStyle = (editor, style) => {
 };
 
 export const isBlockStyleActive = (editor, style) => {
+  const keyName = `style-${style}`;
   const sn = Array.from(
     Editor.nodes(editor, {
-      match: (n) => !Editor.isEditor(n) && typeof n.styleName === 'string',
+      match: (n) => {
+        const isStyle = typeof n.styleName === 'string' || n[keyName];
+        return !Editor.isEditor(n) && isStyle;
+      },
       mode: 'highest',
     }),
   );
 
   for (const [n] of sn) {
-    if (n.styleName.split(' ').filter((x) => x === style).length > 0) {
+    if (typeof n.styleName === 'string') {
+      if (n.styleName.split(' ').filter((x) => x === style).length > 0) {
+        return true;
+      }
+    } else if (
+      n[keyName] &&
+      keyName.split('-').filter((x) => x === style).length > 0
+    )
       return true;
-    }
+    else return false;
```

Comment:
@nileshgulia1 I think this else return will make the for at line 86 useless as it will return after the first iteration no matter what :thinking:

Reply:
@avoinea Nice catch! I noticed the matched elements in the form of `sn` list has always length one. In that case we won't even need a loop. maybe I'm missing some use case here. 

Anyway, `return false` should be outside of loop.